doc/source/whatsnew/v0.25.1.rst

```diff
@@ -128,7 +128,7 @@ Groupby/resample/rolling
 Reshaping
 ^^^^^^^^^
 
--
+- A ``KeyError`` is now raised if ``.unstack()`` is called on a flat :class:`Index` passing a name which is not the correct one (:issue:`18303`)
```

Comment:
"is called on a **Series or DataFrame** with a flat".

Reply:
Seems this affects both stack & unstack?

Reply:
> "is called on a **Series or DataFrame** with a flat".

Fixed, thanks

Reply:
> Seems this affects both stack & unstack?

I added tests for both, but ``stack`` was already raising the error.

Reply:
Understood, thanks.